### PR TITLE
Use CLI parameters instead of config.toml

### PR DESCRIPTION
**Pull request by Luis-omega**
TODO : 
- [x] Add types for parameters.
- [x] Write parser for new cli parameters with compatibility for `--config`  
- [x] Modify main to show a warring if `--config` option is used.
- [x] Show a translation of old toml files to new cli args.
- [x] Delete types for parameters to use only the original Config.
- [x] Add Test. 
- [x] Clean code and format.

### CODE REVIEW

**Comment by t4ccer** [comment]
I don't know if we need to support toml parser, I'd just entirely move to CLI params. We are before V1 so I don't mind breaking changes

**Comment by Luis-omega** [comment]
I was told to write a little pdf that summarizes the change in configuration to users and if I can, to write a script that translates toml config to cli command. I thought that instead of writing a separate script, we can for a time let the user use `ogmios-datum-cache --config "config.toml"` and get directly the right cli command with its config, then after some time, remove the toml parser. Otherwise, I guess I can just copy the current `toml` parser in a new project and tell users "you can use this to translate your config".

**Comment by t4ccer** [comment]
The only place that directly depends on ODC (that I know of) is CTL so it's really a 15 minute change for them

**Comment by Luis-omega** [comment]
In the end, I delete them. 

We need to wait for the merge of  #59  to add the `--origin` option. 

I tried to add quickcheck test for this, but I have too many problems with random strings, so, I left just a fixed test and the isntances.

File: src/Parameters.hs
Comment (t4ccer) [inline src/Parameters.hs:48]: It's hash of block's header, not hash of block

File: src/Config.hs
Comment (t4ccer) [inline src/Config.hs:1]: Maybe move types from this module to `Parameters` to keep them with parsers. After that this module can probably be deleted

**Review by t4ccer — CHANGES_REQUESTED** [review]
Also update README with result of `--help`

File: src/Config.hs
Comment (Luis-omega) [reply src/Config.hs:34]: Yes, I was trying to write a property test, but it's not worth the effort needed, deleting it.

File: src/Config.hs
Comment (Luis-omega) [reply src/Config.hs:1]: Usually, I have a structure like 

```
src 
   | - NameOfThingToParse
             |- Types.hs
             |- Parser.hs
```
But you are right, this is an overkill for the task.

File: src/Parameters.hs
Comment (Luis-omega) [reply src/Parameters.hs]: done.

**Comment by ngua** [comment]
FWIW since I think it's already been decided to totally remove the TOML, I think it will actually actually easier for us to use CLI flags in CTL. At the moment the config.toml is generated using a heredoc in the Arion expression, so it will be more convenient for us to just switch to the CLI flags

**Review by t4ccer — APPROVED** [review]
LGTM, fix tests please before merging

**Pull request merged**
